### ISSUE TITLE
Norwegian number +4749xxxxxx should be valid.

### ISSUE THREAD

**Issue by pongvech** [googlei18n/libphonenumber#1030]
**Country/region affected (e.g., "US"):** NO
**Example number(s) affected ("+1 555 555-1234"):** +4749xxxxxx (http://libphonenumber.appspot.com/phonenumberparser?number=%2B4749xxxxxx)
**The phone number range(s) to which the issue applies ("+1 555 555-XXXX"):** +4749xxxxxx
**The type of the number(s) ("fixed-line", "mobile", "short code", etc.):** mobile
**The cost, if applicable ("toll-free", "premium rate", "shared cost"):** N/A
**Supporting evidence (for example, national numbering plan, announcement from mobile carrier, news article):** http://eng.nkom.no/market/numbering/e.164-numbering-plan/general-norwegian-numbering-plan-for-telephony-etc.e.164

I can't post clients mobile number(Developing application outside Norway). But all numbers with prefix +4749* will fail the validation(isValidNumber).

Expected result : +4749xxxxxx should be valid number.

**Comment by penmetsaa** [comment]
@pongvech  

Thanks for providing all the details!
We've accepted this issue and add 49 range in an upcoming release.

We are unable to find who is the carrier for this new 49 range. 
If possible please provide any official evidence / online reference for same so that we can update carrier name as well.

**Comment by pongvech** [comment]
I think this is the most official evidence we can find: 
http://eng.nkom.no/market/numbering/e.164-numbering-plan/general-norwegian-numbering-plan-for-telephony-etc.e.164

This states: 400 00 000-499 99 999: 8-digit numbers for land mobile services 

In Norway, no carrier owns any specific range of mobile phone numbers. The user is considered the owner. So if a user switches to another carrier, he will keep the number. 

Example:
User buys phone with subscription to Netcom, get number +4740219642
User doesn't like Netcom anymore, so changes to Telenor. 
When contract is signed, Telenor will request a number transfer from Netcom and agree on a date and time (2016-04-01 10:00) when the transfer will take effect. 
User gets new SIM card from Telenor, and a letter that tells him that the card will be activated on 2016-04-01 10:00.
At 2016-04-01 10:00, the transfer will take effect. The old SIM card from Netcom will no longer be able to receive calls. Calls to 40219642 are now routed to Telenor, and to the new SIM.
User changes SIM card in phone and can start using aID again with 4G from Telenor!

So in summary:
4000 00 000-499 99 999 is reserved for mobile phone numbers by the norwegian communication authorities. 
No specific carrier owns the 49* ** *** range, it's available to all mobile carriers.

**Comment by padmaksha** [comment]
Fixed in release 7.3.0
http://libphonenumber.appspot.com/phonenumberparser?number=%2B47+494+00+000

**Comment by gliljas** [comment]
But is it? If it's really true that valid numbers are 400 00 000-499 99 999, then it's not permissive enouigh.

**Comment by penmetsaa** [comment]
@pongvech

Thanks for providing the details.

We understand about number portability in Norway. This is the same case with many other countries. So we always add the original carrier name in these cases. 

If you can find out who's the original carrier, it could help other users of the library support Norwegian users better.


@gliljas

We'll support the rest of the ranges mentioned in nkom.no page. We are tracking it internally.
If there's a specific range that you care about you’re welcome to open a new issue.

**Comment by keghani** [comment]
Re-opening as waiting for a response from @pongvech

**Comment by pongvech** [comment]
In Norway, no carrier owns any specific range of mobile phone numbers, 4000 00 000-499 99 999 are reserved for mobile phone numbers by the norwegian communication authorities.

**Comment by keghani** [comment]
Hi @pongvech,

I don't think that's true. According to http://www.nkom.no/npt/numsys/E.164.pdf many Norwegian mobile numbers have carriers (and we support these in our library for Norway here: https://github.com/googlei18n/libphonenumber/blob/master/resources/carrier/en/47.txt).

That link was updated today actually and according to that the 49 range is blocked. So it seems that our library should indeed fail validation for these numbers. You mentioned you have a client with a 49 number, can you please confirm with them what their number starts with?

**Comment by pongvech** [comment]
Last time I checked we have around 600+ users with 49 from the total users of 400K+ I believe 49 is the new range open. So if you need to know the carrier, what about when user change their carrier but use the same number?

**Comment by pongvech** [comment]
Our system login with phone number, and the registration process need a token in SMS. Now we use workaround for validate +4749xxxxxx, and they continue registration.

**Comment by pongvech** [comment]
I will try to check those numbers again with customer support and get back to you.

**Comment by keghani** [comment]
Thanks @pongvech! I have checked with my team and it seems we're actually aware of some 49 numbers in use. We'll keep supporting 49xxxxxx as valid, so you can remove the workaround on your side once you have the latest version of libphonenumber.

As for carriers, see @penmetsaa's comment above - for regions that support mobile number portability, we return the original carrier. Clients can use [PhoneNumberUtil](https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java)#isMobileNumberPortableRegion(String) to see whether the region has mobile number portability and decide how best to use the carrier mapping in that case.

**Comment by keghani** [comment]
Since http://www.nkom.no/npt/numsys/E.164.pdf is official documentation that was updated today and doesn't have carrier information for 49 I guess we can't do better for this range at this time. So I'll close this issue. Thanks for helping us improve libphonenumber!

**Comment by gliljas** [comment]
FYI. I contacted NKOM directly, and got this response:

> In the Norwegian numbering plan there are a few numbering ranges that are blocked for potential future use, e.g. in case of a need to transfer from a 8-digit numbering plan to a 9-digit numbering plan. Some of the blocked ranges are the 42x, 43x, 44x, 49x and 50x. The use of numbering resources from mandatory numbering plans requires assignment from the Norwegian Communications Authority.  **Numbers from the blocked ranges are not assigned to operators nor end users.** The official Norwegian numbering plan can be found in the link you refer to in your email: http://www.nkom.no/npt/numsys/E.164.pdf.

**Comment by keghani** [comment]
Thanks for checking with the carrier and following up @gliljas, much appreciated!

That's also how I read the numbering plan you referenced, that 49 numbers are not assigned.
But we found some 49 numbers that are actually in use, so we're supporting the range as valid for now.

BTW the corresponding doc in English is at http://www.nkom.no/npt/numsys/E.164_en.pdf.

This is a good topic for the [discussion group](https://groups.google.com/forum/#!forum/libphonenumber-discuss), please consider joining :)

**Comment by gliljas** [comment]
Could you send me (via PM or mail) such a number, so that I can continue my conversation with NKOM and ask how such a number can occur "in the wild"?

I'll take "no" for an answer, but would like to get to the bottom of this.

**Comment by keghani** [comment]
@gliljas Sure, I'd love to keep the team in the loop as well. Would you mind signing [the CLA](https://cla.developers.google.com/about/google-individual)? Then you can be whitelisted to email [libphonenumber-supporting-docs](https://groups.google.com/forum/#!forum/libphonenumber-supporting-docs) and I'll follow up there.

**Comment by gliljas** [comment]
Done 

gunnar.liljas( a )gmail.com

**Comment by penmetsaa** [comment]
Fixed in release 7.3.2: https://libphonenumber.appspot.com/phonenumberparser?number=%2B47+494+00+000

**Comment by keghani** [comment]
For clarification, "Norwegian number +4749xxxxxx should be **in**valid" was fixed in 7.3.2 :)
